dftimewolf/metawolf/utils.py

```diff
@@ -0,0 +1,361 @@
+    tool.LoadConfiguration()
+    try:
+      tool.ReadRecipes()
+    except KeyError:
+      # Recipe already loaded
+      pass
+    except (errors.RecipeParseError,
+            errors.CriticalError) as exception:
+      print(exception)
+      sys.exit(1)
+    return tool
+
+  @staticmethod
+  def CreateNewSessionID() -> str:
+    """Create a new session ID.
+
+    Returns:
+      str: The session ID.
+    """
+    return uuid.uuid4().hex[:6]
+
+  @staticmethod
+  def IsInt(value: str) -> bool:
+    """Check if a string can be safely casted to an int.
+
+    Args:
+      value (str): The string to check.
+
+    Returns:
+      bool: True if the string can be casted to an int.
+    """
+    try:
+      _ = int(value)
+    except ValueError:
+      return False
+    return True
+
+  @staticmethod
+  def IsFloat(value: str) -> bool:
+    """Check if a string can be safely casted to a float.
+
+    Args:
+      value (str): The string to check.
+
+    Returns:
+      bool: True if the string can be casted to a float.
+    """
+    try:
+      _ = float(value)
+    except ValueError:
+      return False
+    return True
+
+  @staticmethod
+  def Str2Bool(value: str) -> Optional[bool]:
+    """Convert a string to its boolean representation.
+
+    Args:
+      value (str): The value to convert to a boolean.
+
+    Returns:
+      Optional[bool]: True/False if the string can be associated to a boolean,
+          None otherwise.
+    """
+    if not isinstance(value, str):
+      return None
+    true_set = {'yes', 'true', 't', 'y'}
+    false_set = {'no', 'false', 'f', 'n'}
+    if value.lower() in true_set:
+      return True
+    if value.lower() in false_set:
+      return False
+    return None
+
+  def GetType(self, value: str) -> Any:
+    """Infer the type of a string's representation.
+
+    Default to str if the underlying string is neither of: [bool, int, float].
+
+    Args:
+      value (str): The string to infer the type from.
+
+    Returns:
+      Any: The type for the string.
+    """
+    if self.Str2Bool(value) in [False, True]:
+      return bool
+
+    if self.IsInt(value):
+      return int
+
+    if self.IsFloat(value):
+      return float
+
+    return str
+
+  def CastToType(self, value: str, value_type: Any) -> Any:
+    """Cast a string to the desired type.
+
+    Returns None if the string cannot be cast to the desired type.
+
+    Args:
+      value (str): The string to cast.
+      value_type (Any): The type to cast the string to.
+    """
+    if value_type == int:
+      if not self.IsInt(value):
+        return None
+      return int(float(value))
+
+    if value_type == float:
+      if not self.IsFloat(value):
+        return None
+      return float(value)
+
+    if value_type == bool:
+      if self.Str2Bool(value) is None:
+        return None
+      return self.Str2Bool(value)
+
+    return value
+
+  @staticmethod
+  def Marshal(st: session.SessionSettable) -> Dict[str, Any]:
```

Comment:
A lot of these are static methods, and the ones above `GetType` and `CastToType` only use `self` to call static methods. Can we extract them from the `MetawolfUtils` class and have them defined at module-level?

Reply:
Certainly can; done.